Menus/MainMenuGUI.cpp

```diff
@@ -1285,12 +1277,13 @@ void MainMenuGUI::Update()
                         }
                         else
                         {
-                            // Save the device type so we can re-set it after resetting - it's really annoying when the device changes
-                            int device = g_UInputMan.GetControlScheme(which - P1CLEAR)->GetDevice();
-                            g_UInputMan.GetControlScheme(which - P1CLEAR)->Reset();
-                            g_UInputMan.GetControlScheme(which - P1CLEAR)->SetDevice(device);
-                            // Set some default mappings so there's something at least
-                            g_UInputMan.GetControlScheme(which - P1CLEAR)->SetPreset(UInputMan::PRESET_NONE);
+                            UInputMan::InputPreset playerPreset = static_cast<UInputMan::InputPreset>(P1CLEAR - which - 1); // Player 1's default preset is at -1 and so on.
+                            UInputMan::Players inputPlayer = static_cast<UInputMan::Players>(which - P1CLEAR);
+                            UInputMan::InputDevice deviceType[4] = { UInputMan::DEVICE_MOUSE_KEYB, UInputMan::DEVICE_KEYB_ONLY, UInputMan::DEVICE_GAMEPAD_1, UInputMan::DEVICE_GAMEPAD_2 };
```

Comment:
Can you reorganize this slightly so it's clearer which things are related?
I'd say what makes most sense is:
```
inputPlayer declaration

playerPreset declaration // maybe rename it defaultPlayerPreset so you can get rid of the comments below
SetPreset(playerPreset)

deviceType declaration // maybe rename it to default input devices so you can get rid of the comment below
SetDevice(deviceType[inputPlayer])
```
Whitespace lines are optional, I put them there to make the pseudocode a little easier to read, I don't know if they'll be helpful or unnecessary in the actual code.

Reply:
i was going with "variables first, actions later" but sure, ok.

Reply:
Fair point. Maybe it's just my unfamiliarity with it, but to me it read like they're all very interconnected, instead of making it clearer which variables apply to which things.

I think this ordering makes it an easier read on skim through, which is generally valuable in a big codebase like this, since it makes it easier to quickly figure out what's going on in a given file/method/whatever.

If you prefer your previous ordering feel free to discuss it, I try to always be open to different ways of looking at things.

Reply:
both ways are equal for me so i'll change it to your suggestion.